### ISSUE TITLE
InvoiceSubscriptionItemOptions missing Price field

### ISSUE THREAD

**Issue by NoahStahl** [stripe/stripe-dotnet#2080]
Hello! Thanks for the excellent library. I'm trying to preview invoices based on add/remove Prices from a subscription, following this doc:

https://stripe.com/docs/billing/subscriptions/prorations#previewing-proration

The example there shows a Price field:

```
// See what the next invoice would look like with a price switch
// and proration set:
var items = new List<InvoiceSubscriptionItemOptions> {
    new InvoiceSubscriptionItemOptions {
        Id = subscription.Items.Data[0].Id,
        Price = "price_CBb6IXqvTLXp3f", // Switch to new price
    },
};
```

However, I don't see this field in InvoiceSubscriptionItemOptions, nor several others documented in the [invoice preview API](https://stripe.com/docs/api/invoices/upcoming?lang=dotnet#upcoming_invoice-subscription_items).

Is there an alternate approach to accomplishing preview with supplied new/removed Price items, or am I missing something? If not, can this field be added to the SDK API? Thanks!

**Comment by NoahStahl** [comment]
I believe I found a workaround, which is using the "Plan" field. Feeding this an ID for a new Price does produce the desired prorated change

```
new InvoiceSubscriptionItemOptions
{
    Plan = pricingId
}
```

For unsubscribing items, you pass the ID of a current SubscriptionItem and flag Delete as true:

```
new InvoiceSubscriptionItemOptions
{
    Id = currentItem.Id,
    Deleted = true
}
```

It would be nice to have the SDK match the expected field names including Price, to avoid confusion. Feel free to close if not wanting to track as feature request. Thanks!

**Comment by remi-stripe** [comment]
@NoahStahl Thanks a lot for the report and sorry for the trouble! You're right that `Plan` works for now though we definitely should support `Price` here and it looks like I missed one case when I did the update.

I'll prepare a PR and get a fix out on Monday!

**Comment by NoahStahl** [comment]
Thanks for the quick reply @remi-stripe! That would be great.

At the risk of being a nuisance, I found another apparently missing SDK field, which is specifying a future start date for a new subscription when previewing its first invoice. The Stripe dashboard issues this query:

`/invoices/upcoming?customer=cus_HRWTdby1MymbCq&subscription_items%5B0%5D%5Bprice%5D=price_1Gv4QpBeli9QhknSAEkdFX0G&subscription_items%5B0%5D%5Bquantity%5D=1&subscription_proration_behavior=create_prorations&subscription_start_date=1593586800`

I'd like to do the same, but not finding anywhere to set `subscription_start_date`. Is there an alternative for this, or can be added?

**Comment by remi-stripe** [comment]
Thanks a lot for catching that one too. Pushed a fix in https://github.com/stripe/stripe-dotnet/pull/2083 and will try to release both together today!

**Comment by remi-stripe** [comment]
@NoahStahl Both fixes were released in 37.12.0!

**Comment by NoahStahl** [comment]
Thanks @remi-stripe for the turnaround! I updated and it looks to be working well.
